### PR TITLE
Add dev process and rc docs

### PR DESCRIPTION
**Pull request by a-blender**
## Problem

 
We have a new TF test process that involves cutting and testing RCs before releasing the rancher2 provider. This needs to be added to the open source docs.

### CODE REVIEW

File: docs/development-process.md
Comment (HarrisonWAffel) [inline docs/development-process.md]: I think it might be useful to split this into two files. What do you think? Maybe, one that community members can reference when preparing to contribute to the repository, and another for Rancher maintainers specifically for releasing it?

File: docs/development-process.md
Comment (HarrisonWAffel) [inline docs/development-process.md]: Could we add a blurb on how to use the testing scripts, contributors might not know how to use it

File: docs/development-process.md
Comment (HarrisonWAffel) [inline docs/development-process.md]: Because this step requires you to have access to Ranchers Zube community members won't be able to follow this. I think leaving the part about Rancher QA validating the issue is still relevant though.

File: docs/development-process.md
Comment (HarrisonWAffel) [reply docs/development-process.md]: I see theres a link to it at the bottom of the file, maybe just reiterating that near the top would help visibility

**Review by HarrisonWAffel — COMMENTED** [review]
In general lgtm, just a few comments regarding the audience for this. Because its in the readme, community members can refer to it before contributing and there are a few steps they won't be able to do as outside contributors

File: docs/development-process.md
Comment (a-blender) [reply docs/development-process.md]: Had offline discussion!

I agree. I split this file into dev process for community members/open source contributors that can also apply to our devs. But, the main reason I want this is to specify that smoke testing features and adding test steps to a GitHub issue is important for our TF maintainers to do our jobs. I/we have to ask for test steps on almost every community PR and docs will help speed up merging PRs.

The `test-process.md` file is for Terraform Maintainers Only and outlines the new release testing process for cutting an RC to test features and bugs.

Let me know if this looks good

File: docs/development-process.md
Comment (HarrisonWAffel) [inline docs/development-process.md]: The list here is stuck at `2`

File: docs/test-process.md
Comment (HarrisonWAffel) [inline docs/test-process.md:5]: Nice! Hopefully this new page will help prevent any confusion on what aspects of the development process are expected of community members versus maintainers

File: docs/test-process.md
Comment (HarrisonWAffel) [inline docs/test-process.md]: These are great examples, I haven't used this script before so having a reference is nice

**Review by HarrisonWAffel — CHANGES_REQUESTED** [review]
This LGTM once we fix the numbered list under the `Development Process`

File: docs/development-process.md
Comment (jakefhyde) [inline docs/development-process.md]: We should instead create a file at `.github/pull_request_template.md` with this content.

File: docs/test-process.md
Comment (jakefhyde) [inline docs/test-process.md]:
```suggestion
```powershell
```

File: docs/test-process.md
Comment (jakefhyde) [inline docs/test-process.md]:
```suggestion
```powershell
```

File: docs/test-process.md
Comment (jakefhyde) [inline docs/test-process.md]:
```suggestion
```sh
```

**Review by jakefhyde — DISMISSED** [review]
Just a few minor nits. Nice job!

**Review by HarrisonWAffel — APPROVED** [review]

**Review by bfbachmann — APPROVED** [review]
Wonderful!

**Comment by daviswill2** [comment]
Closing.

**Pull request merged**
